django/contrib/auth/tests/custom_user.py

```diff
@@ -221,6 +224,31 @@ class CustomUserBadRequiredFields(AbstractBaseUser):
     class Meta:
         app_label = 'auth'
 
+
+class CustomUserUUIDasPK(AbstractBaseUser, PermissionsMixin):
```

Comment:
You can inherit `AbstractUser`, where are all these fields, and only add the `id` field.
It will be generated only if you will not specify it, but if you add it explicitly, it will work as you need.